WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt

```diff
@@ -281,31 +288,45 @@ class WPMainNavigationView @JvmOverloads constructor(
     }
 
     private inner class NavAdapter {
-        private val mFragments = SparseArray<Fragment>(numPages())
+        private val mFragments = mutableMapOf<PageType, Fragment>()
```

Comment:
Do we really need to manually keep references to the fragments? I'm not sure it's safe - what if we unintentionally re-use a destroyed fragment. It won't go though it's regular restoration process and we might end up in an undefined state. I'd personally rather rely on the FragmentManager implementation for the fragment caching, wdyt?

Reply:
Good point! I've removed the map and simplified the adapter. I think now it's much better.